unit-tests/live/memory/test-extrinsics.cpp

```diff
@@ -279,7 +279,9 @@ TEST_CASE("Extrinsic memory leak detection", "[live]")
 
             std::unique_lock<std::mutex> lock(mutex);
             auto pred = [&](){
-                return all_arrived && (new_frame.size() <= cfg_size);
+                if (new_frame.size() > cfg_size)
```

Comment:
I'm not sure you can do this outside of the mutex.
Why not move it to inside the mutex or after the wait?

Reply:
Thanks for pointing this out - fixed.